### PR TITLE
WIP first draft of Python Gitcoin API client

### PR DESCRIPTION
**Pull request by raphaelhuefner**
@mbeacom Please do not merge this PR, but only use it to give me feedback.

Additionally, I have these questions:

- Is it OK to test by mocking the `requests` library so that it does not send any actual HTTP requests? Or do we have to fire up a complete local instance of Gitcoin to run it against?
- I could not really figure out which filter parameters are available for the `https://gitcoin.co/api/v0.1/universe` API endpoint. I only got as far as http://django-filter.readthedocs.io/en/latest/guide/rest_framework.html#using-the-filter-fields-shortcut and that the corresponding `filter_fields` setting is missing from `ExternalBountyViewSet`.
- Could you please clarify who will set up Travis, Pypi and codecov.io ?

Thank you!
Raphael

### CODE REVIEW

**Comment by raphaelhuefner** [comment]
Ah, looks like Travis is already set up.

File: gitcoin/client.py [hunk 1]
```diff
@@ -1,9 +1,13 @@
+"""Define the Gitcoin API client."""
+
 import requests
 
 
 class Config:
     """Define Base Class for API Endpoint Config."""
-    params = {}
```
Comment (mbeacom) [inline gitcoin/client.py]: 👍

**Comment by raphaelhuefner** [comment]
I shouldn't have included tests which run against the live API, I guess... I'll find a way to make them optional, so that Travis won't run them.

At least now we know that my API wrapper will bubble up those 4xx and 5xx HTTP errors. ;-)

**Comment by mbeacom** [comment]
@raphaelhuefner You should be able to use something like `mock` or [`responses`](https://github.com/getsentry/responses) to handle mocking the results from the api.

**Comment by raphaelhuefner** [comment]
This PR is now obsolete, it is superseded by https://github.com/gitcoinco/python-api-client/pull/3

**Pull request closed without merging**